.github/workflows/update_app_templates_hash.yml

```
@@ -0,0 +1,43 @@
+name: Update aws/aws-sam-cli with latest commit hash from aws/aws-sam-cli-app-templates
+
+on:
+  schedule:
+    - cron: "0 * * * *"
+  workflow_dispatch: {}
+
+jobs:
+  updateCommitHash:
+    permissions:
+      pull-requests: write
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout App Templates
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli-app-templates
+          path: aws-sam-cli-app-templates
+
+      - name: Checkout SAM CLI
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli
+          path: aws-sam-cli
+
+      - name: Update commit hash in SAM CLI from App Templates
+        run: |
+          cd aws-sam-cli-app-templates
+          export APP_TEMPLATES_COMMIT_HASH=$(git rev-parse HEAD)
```

nit: `export` is unnecessary AFAICS; the variable is not needed by a subprocess

```suggestion
         APP_TEMPLATES_COMMIT_HASH=$(git rev-parse HEAD)
```